File: medplum-java/medplum-model/src/main/java/com/fhir/Medication.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableMedication.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface Medication {
  Optional<List<Extension>> modifierExtension();

  Optional<List<ResourceList>> contained();

  Optional<Meta> meta();

  Optional<code> status();

  Optional<uri> implicitRules();

  Optional<Ratio> amount();

  Optional<List<Extension>> extension();

  Optional<List<Identifier>> identifier();

  Optional<id> id();

  Optional<code> language();

  Optional<Narrative> text();

  String resourceType();

  Optional<Reference> manufacturer();

  Optional<CodeableConcept> form();

  Optional<Medication_Batch> batch();

  Optional<List<Medication_Ingredient>> ingredient();

  Optional<CodeableConcept> code();

  static ImmutableMedication.ResourceTypeBuildStage builder() {
    return ImmutableMedication.builder();
  }
}

File: medplum-typescript/src/model/fhir/Medication.ts
import { code } from "./code";
import { CodeableConcept } from "./CodeableConcept";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Medication_Batch } from "./Medication_Batch";
import { Medication_Ingredient } from "./Medication_Ingredient";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { Ratio } from "./Ratio";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface Medication {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier[] | null | undefined;
    code: CodeableConcept | null | undefined;
    status: code | null | undefined;
    manufacturer: Reference | null | undefined;
    form: CodeableConcept | null | undefined;
    amount: Ratio | null | undefined;
    ingredient: Medication_Ingredient[] | null | undefined;
    batch: Medication_Batch | null | undefined;
}
